force-app/main/default/classes/GeocodingService.cls

```diff
@@ -0,0 +1,75 @@
+public with sharing class GeocodingService {
+    @InvocableMethod(label='Geocode address')
+    public static List<Coordinates> geocodeAddresses(
+        List<GeocodingAddress> addresses
+    ) {
+        List<Coordinates> computedCoordinates = new List<Coordinates>();
+
+        String baseUrl = 'https://nominatim.openstreetmap.org/search?format=json';
+
+        for (GeocodingAddress address : addresses) {
+            String url = baseUrl;
+            url += (String.isNotBlank(address.street))
+                ? '&street=' + address.street
+                : '';
+            url += (String.isNotBlank(address.city))
+                ? '&city=' + address.city
+                : '';
+            url += (String.isNotBlank(address.state))
+                ? '&state=' + address.state
+                : '';
+            url += (String.isNotBlank(address.country))
+                ? '&country=' + address.country
+                : '';
+            url += (String.isNotBlank(address.postalcode))
+                ? '&postalcode=' + address.postalcode
+                : '';
+
+            Boolean found = false;
+            if (url != baseUrl) {
+                Http http = new Http();
+                HttpRequest request = new HttpRequest();
+                request.setEndpoint(url);
+                request.setMethod('GET');
+                request.setHeader(
+                    'http-referer',
+                    'https://github.com/trailheadapps/dreamhouse-lwc'
```

Comment:
Is a referrer required by the service?
If yes, make this value dynamic with the real org URL.

Reply:
Yes, it's required by the service, changing it to be dynamic